### ISSUE TITLE
IllegalAccessError on BlockPos$MutableBlockPos.<init> (Metalchests or Optifine or BiomesOPlenty)

### ISSUE THREAD

**Issue by thesn10** [asiekierka/FoamFix#233]
**When does the error occur?**
At startup, int the Initialization and Pre-Initialization phase, client-side

**Version information**
minecraft-1.12.2
foamfix-0.10.5-1.12.2
MetalChests-v4.0.1.g01955d6
OptiFine_1.12.2_HD_U_E3
BiomesOPlenty-1.12.2-7.0.1.2427-universal

**This error occurs between:**
-FoamFix and MetalChests
-FoamFix and Optifine
-FoamFix and BiomesOPlenty

If MetalChests and Optifine are installed, Optifine throws the error first.

**1. FoamFix and MetalChests:**
```
java.lang.IllegalAccessError
	at net.minecraft.util.math.BlockPos$MutableBlockPos.<init>(SourceFile:276)
	at net.minecraft.util.math.BlockPos$MutableBlockPos.<init>(SourceFile:271)
	at T145.metalchests.client.gui.GuiHandler.<init>(GuiHandler.java:32)
	at T145.metalchests.core.MetalChests.init(MetalChests.java:165)
        ...
```
[full crash log](https://github.com/asiekierka/FoamFix/files/3541187/crash-metalchests-client.txt)

**2. FoamFix and Optifine:**
```
java.lang.IllegalAccessError
	at net.minecraft.util.math.BlockPos$MutableBlockPos.<init>(SourceFile:276)
	at net.minecraft.util.math.BlockPos$MutableBlockPos.<init>(SourceFile:267)
	at net.minecraft.client.renderer.BlockModelRenderer$AmbientOcclusionFace.<init>(BlockModelRenderer.java:625)
	at net.minecraft.client.renderer.BlockModelRenderer$AmbientOcclusionFace.<init>(BlockModelRenderer.java:616)
	at net.optifine.render.RenderEnv.<init>(RenderEnv.java:43)
        ...
```
[full crash log](https://github.com/asiekierka/FoamFix/files/3541188/crash-optifine-client.txt)

**3. FoamFix and BiomesOPlenty:**
```
java.lang.IllegalAccessError
	at net.minecraft.util.math.BlockPos$MutableBlockPos.<init>(SourceFile:276)
	at net.minecraft.util.math.BlockPos$MutableBlockPos.<init>(SourceFile:267)
	at net.minecraft.world.biome.Biome.func_180628_b(Biome.java:250)
	at net.minecraft.world.biome.Biome.func_180622_a(Biome.java:230)
	at biomesoplenty.common.world.ChunkGeneratorOverworldBOP.replaceBlocksForBiome(ChunkGeneratorOverworldBOP.java:325)
	at biomesoplenty.common.world.ChunkGeneratorOverworldBOP.func_185932_a(ChunkGeneratorOverworldBOP.java:148)
        ...
```
[full crash log](https://github.com/asiekierka/FoamFix/files/3541393/crash-biomesoplenty-server.txt)

FoamFix has the same incompatiblility with two different mods, which seems to be related to whatever foamfix is doing with the MutableBlockPos class.

**Comment by thesn10** [comment]
Doesnt crash if i disable "optimizedBlockPos" in the config.

**Comment by asiekierka** [comment]
May be fixed in 0.10.6.
